File: src/dotnet/Ansight/Pairing/Models/PairingTicket.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Self-contained pairing payload used by file, QR, and bundled pairing flows.
/// </summary>
public sealed class PairingTicket
{
    /// <summary>
    /// Schema identifier for pairing ticket payloads.
    /// </summary>
    public const string SchemaName = "ansight.pairing-ticket.v1";

    /// <summary>
    /// Schema identifier for this ticket.
    /// </summary>
    public string Schema { get; set; } = SchemaName;

    /// <summary>
    /// Signed pairing config carried by the ticket.
    /// </summary>
    public required PairingConfig Config { get; set; }

    /// <summary>
    /// Discovery metadata that helps the client reach the host.
    /// </summary>
    public PairingDiscoveryHint? Discovery { get; set; }
}

File: src/ios/Sources/AnsightCore/PairingTicket.swift
import CryptoKit
import Foundation

public struct PairingTicket: Sendable, Codable, Equatable {
    public static let schemaName = PairingConfigDocument.legacySchemaName

    public var schema: String
    public var config: PairingConfig
    public var discovery: PairingDiscoveryHint?

    public init(
        schema: String = PairingTicket.schemaName,
        config: PairingConfig,
        discovery: PairingDiscoveryHint? = nil
    ) {
        self.schema = schema
        self.config = config
        self.discovery = discovery
    }
}
